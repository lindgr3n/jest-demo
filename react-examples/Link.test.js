// Link.react-test.js
import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Enzyme variant', () => {
    const component = shallow(
        <Link page="http://www.facebook.com">Facebook</Link>
    );
    const alink = component.find('a')
    expect(component).toMatchSnapshot()
    
    alink.simulate('mouseEnter')
    expect(component).toMatchSnapshot()
    alink.simulate('mouseLeave')

    expect(alink.prop('className')).toEqual('normal')
})

it('Empty link', () => {
    const component = shallow(
        <Link>Undefined</Link>
    );
    const alink = component.find('a')
    expect(alink.prop('href')).toEqual('#')
})