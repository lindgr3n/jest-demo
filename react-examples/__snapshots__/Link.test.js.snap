// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme variant 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <a
    className="normal"
    href="http://www.facebook.com"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
>
    Facebook
</a>,
  "nodes": Array [
    <a
      className="normal"
      href="http://www.facebook.com"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
>
      Facebook
</a>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Link
        page="http://www.facebook.com"
>
        Facebook
</Link>,
      "_debugID": 4,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Link {
        "_onMouseEnter": [Function],
        "_onMouseLeave": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": "Facebook",
          "page": "http://www.facebook.com",
        },
        "refs": Object {},
        "state": Object {
          "class": "normal",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <a
          className="normal"
          href="http://www.facebook.com"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          Facebook
</a>,
        "_debugID": 5,
        "_renderedOutput": <a
          className="normal"
          href="http://www.facebook.com"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          Facebook
</a>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Link
    page="http://www.facebook.com"
>
    Facebook
</Link>,
}
`;

exports[`Enzyme variant 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <a
    className="hovered"
    href="http://www.facebook.com"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
>
    Facebook
</a>,
  "nodes": Array [
    <a
      className="hovered"
      href="http://www.facebook.com"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
>
      Facebook
</a>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Link
        page="http://www.facebook.com"
>
        Facebook
</Link>,
      "_debugID": 4,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Link {
        "_onMouseEnter": [Function],
        "_onMouseLeave": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": "Facebook",
          "page": "http://www.facebook.com",
        },
        "refs": Object {},
        "state": Object {
          "class": "hovered",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <a
          className="hovered"
          href="http://www.facebook.com"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          Facebook
</a>,
        "_debugID": 5,
        "_renderedOutput": <a
          className="hovered"
          href="http://www.facebook.com"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          Facebook
</a>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Link
    page="http://www.facebook.com"
>
    Facebook
</Link>,
}
`;

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
